src/views/GameView.vue

```diff
@@ -702,6 +699,11 @@ export default {
     },
   },
   mounted() {
+    if (this.isSpectating && !this.$store.state.game.id) {
+      const { gameId } = this.$router.currentRoute.value.params;
+      this.$store.dispatch('requestSpectate', Number(gameId));
```

Comment:
Same here, this may not be defensive enough since it will check any game id from the route params.

Reply:
Correct. Not expected, but maybe not impossible.

Reply:
Thoughts on what the UX should be to handle this error? I think we need to navigate home if that happens. The same would apply in the GameView itself when playing